MBS-13723: Allow "download"-only relationships to Qobuz URLs

yvanzo opened the pull request:
<!--
    Hello! Thanks for submitting a pull request to MusicBrainz Server.
    We appreciate your time and interest in helping our project!

    Please use this template to help us review your change.

    Depending on your change, some sections may be unneeded, just remove these.
    For example, small pull requests usually don’t need the section “Action”.

    Remember that the more helpful info your pull request includes,
    the easier it is for us to understand and review your changes.

    Ensure that you’ve read through and followed the Contributing Guidelines, at
    https://github.com/metabrainz/musicbrainz-server/blob/master/CONTRIBUTING.md
-->

# Problem MBS-13723
<!--
    Anything that helps us understand why you are making this change goes here.
    What problem are you trying to fix? What does this change address?
-->

When implementing MBS-13438 it was thought that Qobuz always has a streaming option, but some releases are available for download only which isn’t currently allowed in the external links editor.


# Solution
<!--
    Talk about technical details, considerations, or other interesting points.
-->

Just allow "download"-only additionally to the already allowed "streaming"-only (auto-selected) and "download"+"streaming".

Additionally, now that download-only is an option, the streaming relationship type is no longer a subset of all possible options. It would be misleading to keep auto-selecting the streaming relationship type for Qobuz URLs. Therefore this patch also disables auto-selecting relationship type for Qobuz URLs.



# Testing
<!--
    Talk about the testing you have or haven’t done (whether you rely
    on automated tests, or don’t know how to test something). It’s useful
    for others to know what you’ve already tested so that they can avoid
    repeating the same steps, and instead consider what you might’ve
    missed.
-->

Tested locally in a release editor and updated CI tests.

reosarevok reviewed (commented):
Should streaming still be preselected then, if some releases cannot be streamed?

yvanzo commented:
Thank you @reosarevok, I added a commit to disable relationship type auto-selection and to update tests accordingly. I updated also the description of the pull request and of the ticket.

reosarevok reviewed (approved):
LGTMBDNT
